### ISSUE TITLE
Inheritance of templates in Helm

### ISSUE THREAD

**Issue by dgawlik** [helm/helm#13398]
Hello, 

I have a more generic kind of question. First I will try to describe what I'm trying to achieve

### Context

Lately I viewed some presentation about Radius CNCF incubating project. There was this cool concept of implementing of K8s application graphs of dependencies. So I thought to myself that I will test it by adding k8s annotations to deployments like `dependsOn` and `dependsOnPort` for every component of the system and then use kubernetes client and graphviz to paint a graph of dependencies. 

### Problem

So I have for example bitnami/posgres and bitnami/redis and I'm not seeing an easy way to to extend their manifests without reimplementing the charts from scratch. What would be nice in Helm and I think is missing is some kind of inheritance. If I got documentation right then there is this concept of parent and subchart but it only relates to overriding values not the manifest structure. So it would be cool to reuse already defined helm charts, somehow inheriting the manifests and then "injecting" your values in custom places. Like for my case I would take manifests as is and in extending chart I would add couple of annotations on top.

### Proposal

I thought if there was a way to define `parentChart` and `parentChartVersion` in chart you would import all the yamls. Then you would have overrides directory (similar to templates) and there you would put your overrides. So for example

```
meta:
    overridesFor: DEPLOYMENTS | PODS | SERVICES
extends:
   meta:
      annotations:
         annotation1: "annotationValue1"
         annotation2: "annotationValue2"
```

### Discussion

The above was only a proposition. I just wanted to initiate the discussion around this particular topic. On the other hand maybe I didn't get everything and maybe some more experienced person will show me how to achieve the above with what already is.

**Comment by banjoh** [comment]
If I understand your proposal here, you would like to have helm patch k8s resources of a certain type with annotations, right?

Regardless of how such a feature were to be added to helm, we would still lack a way to annotate resources created by operators. Annotating such resources can only be done by chart authors.

**Comment by dgawlik** [comment]
Yes that's exactly the thing. It seems silly to reimplement 10+ manifests for such simple change, apart from this -  from experience I know that helm authors not always get all the things you want them.

This is hypothetical example only, I don't need it for commercial purposes.

It's just that I know that docker (and other container runtimes) used the overlayfs and layered architecture to much of the success. So here is the same thing basically the manifest/patch on top wins in merged view. I think it would be cool to see something like this in couple of years (who knows). 

But on the flipside I suppose that it would be a bit of a hell to implement regarding how k8s library is designed. But maybe not - you tell me would it be hard? (in theory). I was thinking of making my feet wet and maybe scribbling some kind of poc on my fork. But at least I must know that the idea is not going straight to trash, since why bother.

On the other hand maybe you have an other idea how to patch that other way (semi-automatically)? Maybe with Kustomize help?
